### DISCUSSION TITLE
Pbs with filters

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#2763] [General]
Hello,

I'm working on a mapping application in which I have a JSON style loaded with points representing breweries.
I declare my source (in this case the style), my layer. No problem.
This application requires authentication (login, password, sessions...). Each identified user has authorized communes (e.g. user 1 will have Paris as authorized commune, and another Berlin etc.). I retrieve a session variable which is a JSON object with all authorized commune codes.
I rework the code provided so that it fits into the "filter" field of map.addLayer...

```
communes_aut = sess_communes.toString();

communes_join = '\''+sess_communes.join('\',\'')+'\'';

communes_filter = communes_join.toString();

console.log(communes_filter)
// return '82032','82112','82170'

```

Here's the problem: when I enter this variable in the "filter" field, it doesn't work, but when I enter the hard data instead, it does....

That doesn't work : 
```
 var brw_filter=[
    "all",
    ["in", "code_commune", communes_filter],
    ["in", "code_pays", 'FRA']
]
  map.setFilter('brw_layer',brw_filter)

```

That works !!

```
  var brw_filter=[
    "all",
    ["in", "code_commune",  '82032','82112','82170'],
    ["in", "code_pays", 'FRA']
]
  map.setFilter('brw_layer',brw_filter)


```

Why this problem and is there a solution ?

Thanks

Sylvain

**Comment by HarelM** [comment]
Looks like a bad string...? Make sure to use typescript...

**Comment by sylvain258** [comment]
@HarelM It's possible,

This is the content of my session variable :
`["82032","82112","82170"]`

It's a valid JSON, confirmed by a online tool.

After, I have no idea to convert this data in compatible data for Maplibre filter.

Anyone would have an idea ? 

Thanks

Sylvain

**Comment by sylvain258** [comment]
So, after many difficulties, here is a partial solution:
`
  var brw_filter=[
    "any",
    ["in", ["get", "code_commune"], ["literal", sess_communes]]];
 //   ["in", ["get", "code_pays"], 'FRA']];`

Partial solution, because I'm still trying to figure out how to combine several filters (second line of the comment).

Thks

Sylvain
